lib/mayaUsd/nodes/proxyShapeBase.cpp

```diff
@@ -1589,7 +1589,7 @@ bool MayaUsdProxyShapeBase::isStageIncoming() const
 
         // Check what is the cache connected to
         MStringArray result;
-        MGlobal::executeCommand(("listConnections -shapes on " + this->name()), result);
+        MGlobal::executeCommand(("listConnections -t shape -shapes on " + this->name()), result);
 
```

Comment:
Is there any reason why you did not choose to go with the solution that was suggested on github? I would like to understand why that solution was not correct? The suggestion was:
```
bool isIncomingStage = false;
if (!inDataHandle.data().isNull()) {
    isIncomingStage = true;
} 
```

Reply:
Hi @pierrebai-adsk , the reason is stated in the description. So if we remove the `else` part and leave it like this it will not work properly for Bifrost. For some reason (I get the conclusion from debugging),  inDataHandle can still be Null even if we have Bifrost shape connected to it. In that case, we need to check the stage Cache and find the Bifrost shape in Cache. Hence the `else` part.